File: src/3000.maximum-area-of-longest-diagonal-rectangle.3251/solution.ts
function areaOfMaxDiagonal (dimensions: number[][]): number {
    let res = 0;
    let max = 0;
    for (const [x, y, ] of dimensions) {
        const d = x ** 2 + y ** 2;
        if (d > max) {
            max = d;
            res = x * y;
        } else if (d === max) {
            res = Math.max(res, x * y);
        }
    }
    return res;
}

File: src/3000.maximum-area-of-longest-diagonal-rectangle.3251/solution.java
class Solution {
    public int areaOfMaxDiagonal(int[][] dimensions) {
        int area = 0;
        int diagonal = 0;
        for(int[] rectangle : dimensions){
            int d = rectangle[0]*rectangle[0]+rectangle[1]*rectangle[1];
            if(d>diagonal || (d == diagonal && rectangle[0]*rectangle[1]>area)){
                diagonal = d;
                area = rectangle[0]*rectangle[1];
            }
        }
        return area;
    }
}